bip-0183.md

```diff
@@ -0,0 +1,570 @@
+This BIP defines the networking-layer changes needed to allow nodes to exchange the inclusion proofs.
+This document **does not** describe how to validate blocks and transactions using the provided data, see [BIP-0182](bip-0182.md) for more details.
+
+## Motivation
+
+Utreexo nodes require the inclusion proof to fully validate blocks and transactions.
+Each block has a corresponding inclusion proof with it and this inclusion proof for blocks up to height 906,937 requires an additional 631.85GB, which is roughly 40GB less than the size of the block data.
+Each transaction also has a corresponding inclusion proof with it and for normal transaction relay, the proof is roughly 3 times the size of the transaction.
+It's still reasonable for a single node to download this extra data but a little caching goes a long way in reducing the amount of data that one has to download.
+We define the new P2P messages for the inclusion proofs to support caching to reduce bandwidth while also allowing a high bandwidth, low-latency usage.
+
+## License
+
+This BIP is licensed under the BSD-3-Clause license.
+
+## Overview
+
+### Requirements and Compatibility
+
+Nodes implementing Utreexo can choose which messages to support.
+There are a number of configurations possible, and this BIP does not restrict nodes to any subsets of messages.
+
+That said, there are three likely types of nodes:
+1. Compact State Nodes (CSNs)
+2. Bridge nodes
+3. Archive nodes
+
+CSNs have the goal of minimizing data storage and download while performing block validation.
+Archive and bridge nodes store more data and provide this data to CSNs.
+
+Bridge nodes are nodes that can add inclusion proofs to mempool transactions, support the same set of messages as CSNs, and should in fact be indistinguishable from CSNs on the network.
+Archive nodes are able to serve the blocks and the inclusion proofs. However, they are not able to generate the inclusion proofs as they do not keep the full UTXO set.
+
+Note that the archive and bridge capabilities of a node are separate; a bridge node can be bridge only, without previous block proof data, and an archive node doesn't need to be able to bridge.
+
+The one exception to this flexibility is that archive nodes must provide both the blocks and the inclusion proofs.
+While theoretically possible to split these two resources, the blocks are quite small relative to the block proofs, and it simplifies clients to be able to rely on being able to request both over the same connection.
+
+### Pre-P2P: Bridge Building
+
+When introducing Utreexo into an existing network, there are two things needed before CSNs can operate.
+First, archive nodes need to build proofs for old blocks to serve during the initial block download (IBD).
+Second, nodes need to build and maintain the UTXO merkle forest, and an index of outpoints to leaves of that forest, so that they can build proofs for new transactions.
```

Comment:
For some reason, "Merkle" is consistently capitalized in 181 and 182 but consistently lowercase in 183.